core/dive.cpp

```diff
@@ -1148,12 +1148,12 @@ static void merge_one_sample(const struct sample &sample, struct divecomputer &d
 
 			/* Init a few values from prev sample to avoid useless info in XML */
 			surface.bearing.degrees = prev.bearing.degrees;
-			surface.ndl.seconds = prev.ndl.seconds;
+			surface.ndl = prev.ndl;
 			surface.time.seconds = last_time + 20;
 
 			append_sample(surface, &dc);
 
-			surface.time.seconds = sample.time.seconds - 20;
+			surface.time -= duration_t { .seconds = 20 };
```

Comment:
I think this is wrong, as it removes `sample.time` from the assignment.

Reply:
Yes, I misread sample as surface.